features/networking/pod.feature

```diff
@@ -190,3 +190,124 @@ Feature: Pod related networking scenarios
+    
+    Given I select a random node's host
+    And I have a project
+    #pod-for-ping will be a non-hostnetwork pod
+    And I have a pod-for-ping in the project
+    And the pod named "hello-pod" becomes ready
+    #Curl on Master's tun0 IP to make sure connections are blocked to MCS via tun0
+    When I execute on the pod:
+      | curl | -I | http://<%= cb.master_tun0_ip %>:22623/master/config | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | http://<%= cb.master_tun0_ip %>:22624/master/config | -k |
+    Then the output should contain "Connection refused"
+
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23892
+  @admin
+  Scenario: A pod cannot access the MCS via an egress router in http proxy mode
+    Given I use the first master host
+    #Step to obtain master IP
+    And I run commands on the host:
+      | hostname -i |
+    Then the step should succeed
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
+    Given I have a project
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    #pod-for-ping will be a non-hotnetwork pod
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/weliang1/Openshift_Networking/master/egress-http-proxy/egress-http-proxy-pod.yaml |
+      | n | <%= project.name %>                                                                                   |
+    Then the pod named "egress-http-proxy" becomes ready
+    #Pod should not access MCS via an egress router pod
+    When I execute on the pod:
+      | curl | -I | http://<%= cb.master_ip %>:22623/master/config | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | http://<%= cb.master_ip %>:22624/master/config | -k |
+    Then the output should contain "Connection refused"
+
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23893
+  @admin
+  Scenario: A pod in a namespace with an egress IP cannot access the MCS
+    Given I use the first master host
+    #Step to obtain master IP
+    And I run commands on the host:
+      | hostname -i |
+    Then the step should succeed
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
+    Given I have a project
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    #pod-for-ping will be a non-hostnetwork pod
+    And I have a pod-for-ping in the project
+    And the pod named "hello-pod" becomes ready
+    
+    When I run the :patch admin command with:
+      | resource      | netnamespace                         |
+      | resource_name | <%= project.name %>                  |
+      | p             | {"egressIPs":["<%= cb.valid_ip %>"]} |
```

Comment:
As you said, the `valid_ip` is needed, you can refer to the related step from `egress-ip.feature` feature.

Reply:
Thats right. Thanks @zhaozhanqi , i was expecting the similar recommendation from you. Let me incorporate it and test with that

Reply:
So i had to change networking.rb to add --type=merge in steps. Works fine now. Since this rb change is going to be only in master branch, i am not sure if we need this change is v3 branch too?

Reply:
Ok so we don't need to make this change in v3 branch. Lets keep v3 code as as. Tjis PR is going to merge in master so it won;t impact v3 src code